spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1579,19 +1579,24 @@ def paste(self):
         text has 'CRLF' EOL chars
         """
         clipboard = QApplication.clipboard()
-        # This is here to prevent pasting mime data in the Editor.
-        # See issue 8566 for the details.
+        text = to_text_string(clipboard.text())
+        eol_chars = self.get_line_separator()
         if clipboard.mimeData().hasUrls():

```

Comment:
@Khalilsqu Is your comment on the previous PR still valid?

> In windows, some application even format text data as url, and my testing shows sometimes I cannot paste codes from pycharm to spyder if only hasUrls() is used.

I haven't been able to reproduce it myself with URLs from web browsers on Windows, but preventing pasting entirely from certain applications is a pretty serious problem. If so, then we should try to resolve that issue by, e.g. checking that the URLs in the clipboard all start with ``file:///``.

Reply:
It is fixed, with this line `if all([url.isLocalFile() for url in urls]):`, text will not be modified except if the copied files are indeed files (this works similar to `osp.exists()`.

Reply:
Great, perfect!